shared-libs/transitions/src/transitions/conditional_alerts.js

```diff
@@ -19,20 +19,15 @@ const evaluateCondition = (doc, alert) => {
   if (alert.condition.indexOf(alert.form) === -1) {
     return runCondition(alert.condition, context);
   }
-  return utils.getReportsWithSameClinicAndForm({
-    doc: doc,
-    formName: alert.form
-  })
-    .then(rows => {
-      rows = _.sortBy(rows, function(row) {
-        return row.doc.reported_date;
-      });
-      context[alert.form] = function(i) {
-        const row = rows[rows.length - 1 - i];
-        return row ? row.doc : row;
-      };
-      return runCondition(alert.condition, context);
-    });
+  return utils.getReportsWithSameClinicAndForm({ doc: doc, formName: alert.form })
+    .then(rows => rows.map(row => row.doc))
+    .then(docs => docs.sort((lhs, rhs) => lhs.reported_date > rhs.reported_date))
+    .then(docs => {
+      docs.push(doc);
```

Comment:
What I would rather do is:
- before any kind of sorting, check if the current doc exists in the results list
- if exists, replace the entry in the results list with the current doc (in case another transition has edited it in a relevant way)
- if it doesn't exist, push it
- sort and map to the context function, descending by reported_date
What do you think?

Reply:
Oh you're right! This way may end up with two copies of the same doc which breaks the indexing. Thanks!